test/invariants.js

```diff
@@ -0,0 +1,50 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('invariants', function() {
+
+    it('-- every non-nullary function throws if given no arguments', function() {
+        var isNoArgsException = function(err) {
+            return (
+                err.constructor === TypeError &&
+                err.message === 'Function called with no arguments'
+            );
+        };
+
+        for (var prop in R) {
+            if (prop !== 'lazylist' && typeof R[prop] === 'function' && R[prop].length > 0) {
+                assert.throws(R[prop], isNoArgsException);
+            }
+        }
+    });
+
+    it('-- applying function f with length n (where n > 0) to R.__ gives function with length n', function() {
+        for (var prop in R) {
+            if (prop !== 'lazylist' && typeof R[prop] === 'function' && R[prop].length > 0) {
+                var result = R[prop](R.__);
+                assert.strictEqual(typeof result, 'function');
+                assert.strictEqual(result.length, R[prop].length);
+            }
+        }
+    });
+
+    it('-- applying function f with length n (where n > 1) to any value other than R.__ gives function with length n - 1', function() {
```

Comment:
This really does nothing like testing 'any other value than `R.__`', but unless we get into property-based testing, we can't really do that.